### PR TITLE
add (and use) gcd_with_cofactor in ref_ff

### PR DESCRIPTION
**Pull request by fieker**

### CODE REVIEW

**Comment by thofma** [comment]
@tthsqe12 will move this to AA/Nemo

**Comment by tthsqe12** [comment]
However, if you need it in oscar yesterday, i would recommend using the wrong name as in the title.

**Comment by fieker** [comment]
On Wed, Nov 16, 2022 at 10:34:46AM -0800, Tommy Hofmann wrote:
> @tthsqe12 will move this to AA/Nemo
And add the s at the end...
Thanks

**Comment by thofma** [comment]
I removed the methods since they now live in AA/Nemo. But it seems they are not used (yet)?

**Comment by fieker** [comment]
On Sat, Nov 19, 2022 at 09:08:05AM -0800, Tommy Hofmann wrote:
> I removed the methods since they now live in AA/Nemo. But it seems they are not used (yet)?
> 
There are used in exactly that file in some rref/ref function written
for Mohammed
They should be used "everywhere" as the modular gcd tries division to
test completion, but multiplication by the fofactors would be faster,
much faster...

File: experimental/ModStd/ModStdQt.jl [hunk 1]
```diff
@@ -499,12 +521,8 @@ function ref_ff_rc!(M::MatElem{<:MPolyElem})
       if iszero(M[k, j])
         continue
       end
-      g = gcd(M[k, j], M[i, j])
-      if isone(g)
-        M[k, :] = M[i, j]*M[k, :] - M[k, j] * M[i, :]
-      else
-        M[k, :] = divexact(M[i, j], g)*M[k, :] - divexact(M[k, j], g) * M[i, :]
-      end
+#      g, a, b = gcd_with_cofactors(M[k, j], M[i, j])
```
Comment (thofma) [inline experimental/ModStd/ModStdQt.jl:524]: I only saw this line, which is a comment and the `a,b` below are not defined.

**Review by fieker — COMMENTED** [review]
i'll look at the rref code soon, but merge this as is

**Pull request merged**